src/serena/agent.py

```diff
@@ -603,21 +605,74 @@ def mark_file_modified(self, relative_path: str) -> None:
         assert self.lines_read is not None
         self.lines_read.invalidate_lines_read(relative_path)
 
-    def __del__(self) -> None:
-        """
-        Destructor to clean up the language server instance and GUI logger
-        """
-        if not hasattr(self, "_is_initialized"):
-            return
-        log.info("SerenaAgent is shutting down ...")
+    def __enter__(self) -> "SerenaAgent":
```

Comment:
SerenaAgent is never used with `with`, and I don't see why it should. Do you actually need this for a more graceful shutdown?